FXyz-Importers/src/test/java/org/fxyz3d/importers/obj/ObjImporterTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * F(X)yz
+ *
+ * Copyright (c) 2013-2019, F(X)yz
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *     * Neither the name of F(X)yz, any associated website, nor the
+ * names of its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL F(X)yz BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.fxyz3d.importers.obj;
+
+import org.fxyz3d.importers.Model3D;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Almas Baimagambetov (almaslvl@gmail.com)
+ */
+class ObjImporterTest {
+
+    private ObjImporter importer;
+
+    @BeforeEach
+    void setUp() {
+        importer = new ObjImporter();
+    }
+
+    @Test
+    void testExtensions() {
+        assertTrue(importer.isSupported("obj"));
+    }
+
+    @Test
+    void testLoad() throws Exception {
+        Model3D model = importer.load(getClass().getResource("cube.obj"));
+
+        assertEquals(1, model.getRoot().getChildren().size());
+        assertEquals(1, model.getMeshNames().size());
+        assertTrue(model.getMeshNames().contains("cube"));
+        assertSame(model.getMeshView("cube"), model.getRoot().getChildren().get(0));
+    }
```

Comment:
I think we want some check for the content of `mesh.getPoints()`, `mesh.getTexCoords()`, `mesh.getFaces()`, `mesh.getNormals()`... These are the fundamental properties of the model and we would want to make sure they are read correctly. Could have a separate method for each check.

Reply:
This is a good start.

Obviously we can add more tests as we go. 

For instance, we should add a material and a `mtl` file, look for a more complex obj, ...

The `TriangleMesh` has already internal checks so I don't think we should test if the model from the obj itself is a proper mesh, but if we _do_ read it correctly.

Reply:
added mtl test and multi view obj file. Not sure about poly faces yet, as I don't know what we agreed on wrt loadAsPoly

Reply:
You can test an obj file with 3 or more vertices per face using loadAsPoly. For now, we let the user decide how to import the file.

Reply:
ok